src/index.ts

```diff
@@ -474,46 +366,46 @@ class WorkerInfo extends AsynchronouslyCreatedResource {
     port : MessagePort,
     onMessage : ResponseCallback) {
     super();
-    this.worker = worker;
+    this[kWorkerWorkerThread] = worker;
     this.port = port;
     this.port.on('message',
-      (message : ResponseMessage) => this._handleResponse(message));
+      (message : ResponseMessage) => this.#handleResponse(message));
     this.onMessage = onMessage;
     this.taskInfos = new Map();
-    this.sharedBuffer = new Int32Array(
+    this[kWorkerSharedBuffer] = new Int32Array(
       new SharedArrayBuffer(kFieldCount * Int32Array.BYTES_PER_ELEMENT));
   }
 
   destroy () : void {
-    this.worker.terminate();
+    this[kWorkerWorkerThread].terminate();
     this.port.close();
-    this.clearIdleTimeout();
+    this.#clearIdleTimeout();
     for (const taskInfo of this.taskInfos.values()) {
       taskInfo.done(Errors.ThreadTermination());
     }
     this.taskInfos.clear();
   }
 
-  clearIdleTimeout () : void {
+  #clearIdleTimeout () : void {
```

Comment:
Wouldn't it make this change a semver-major? I think you can do it in a separate PR

Reply:
Agree, but AFAIT we do not expose those internals isn't it?

If so, we should make it; otherwise I believe they can go under a minor, specially as we are not changing the behaviour by default, let me know what you think 🤔

Reply:
I'm not 100% familiar with this API. If you say we don't expose it, that's fine.

Reply:
Thanks! Yeah, I double checked, we do not expose them, are meant for internal usage 👍